realm/realm-library/src/androidTest/java/io/realm/RealmAsyncQueryTests.java

```diff
@@ -1104,23 +1105,36 @@ public void badVersion_syncTransaction() throws NoSuchFieldException, IllegalAcc
         result.addChangeListener(new RealmChangeListener<RealmResults<AllTypes>>() {
             @Override
             public void onChange(RealmResults<AllTypes> object) {
-                // 4. The commit in #2, should result in a refresh being triggered, which means this callback will
-                // be notified once the updated async queries has run.
                 assertTrue(result.isValid());
                 assertTrue(result.isLoaded());
-                assertEquals(1, result.size());
-                looperThread.testComplete();
+                switch (listenerCount.getAndIncrement()) {
+                    case 0:
+                        // Triggered by beginTransaction
+                        assertEquals(0, result.size());
+                        break;
+                    case 1:
+                        // 4. The commit in #2, should result in a refresh being triggered, which means this callback will
+                        // be notified once the updated async queries has run.
+                        assertEquals(1, result.size());
+                        looperThread.testComplete();
+                        break;
+                    default:
+                        fail();
+                        break;
+                }
             }
         });
 
         // 2. Advances the caller Realm, invalidating the version in the handover object.
         realm.beginTransaction();
+        assertTrue(result.isLoaded());
         realm.createObject(AllTypes.class);
         realm.commitTransaction();
 
         // 3. The async query should now (hopefully) fail with a BadVersion.
```

Comment:
I put a `NOTE` on top of this test:
>     // NOTE: This test is not checking the same thing after the OS results integration. Just keep it for an additional test for async.
To make it less confusing, i decided to keep the original comments there :P